annotation-processor/src/main/java/org/hibernate/validator/ap/checks/ConstraintCheckFactory.java

```diff
@@ -115,13 +154,19 @@ public ConstraintCheckFactory(Types typeUtils, ConstraintHelper constraintHelper
 						new AnnotationParametersPatternCheck( annotationApiHelper ),
 						new AnnotationParametersScriptAssertCheck( annotationApiHelper ),
 						new AnnotationParametersDigitsCheck( annotationApiHelper ),
-						new AnnotationParametersDecimalMinMaxCheck( annotationApiHelper )
+						new AnnotationParametersDecimalMinMaxCheck( annotationApiHelper ),
+						new MethodOverrideCheck( elementUtils, typeUtils )
 				)
 		);
 		methodChecks.put(
 				AnnotationType.GRAPH_VALIDATION_ANNOTATION,
-				new SingleValuedChecks( new GetterCheck( methodConstraintsSupported ), new StaticCheck(), new MethodAnnotationCheck( constraintHelper ),
-						new PrimitiveCheck() )
+				new SingleValuedChecks(
+						new GetterCheck( methodConstraintsSupported ),
+						new StaticCheck(),
+						new MethodAnnotationCheck( constraintHelper ),
```

Comment:
I'm wondering why we apply `MethodAnnotationCheck` to `@Valid` (I know, it has been here before). This seems not correct.